README.md

```diff
@@ -142,6 +142,87 @@ To change this behavior use `skipOnEmpty(true)`:
 Number::rule()->integer()->max(100)->skipOnEmpty(true)
 ```
 
+#### Nested and related data
+
+In many cases there is a need to validate related data in addition to current entity / model. There is a `Nested` rule 
+for this purpose:
+
+```php
+use Yiisoft\Validator\Rule\HasLength;
+use Yiisoft\Validator\Rule\Nested;
+use Yiisoft\Validator\Rule\Number;
+use Yiisoft\Validator\Rule\Required;
+
+$data = ['author' => ['name' => 'Alexey', 'age' => '31']];
+$rule = Nested::rule([
+    'title' => [Required::rule()],
+    'author' => Nested::rule([
+        'name' => [HasLength::rule()->min(3)],
+        'age' => [Number::rule()->min(18)],
+    )];
+]);
+$errors = $rule->validate($data)->getErrors();
+```
+
+A more complex real-life example is a chart that is made of points. This data is represented as arrays. `Nested` can be 
+combined with `Each` rule to validate such similar structures:
+
+```php
+use Yiisoft\Validator\Rule\Each;
+use Yiisoft\Validator\Rule\Nested;
+
+$data = [
+    'charts' => [
+        [
+            'points' => [
+                ['coordinates' => ['x' => -11, 'y' => 11], 'rgb' => [-1, 256, 0]],
+                ['coordinates' => ['x' => -12, 'y' => 12], 'rgb' => [0, -2, 257]]
+            ],
+        ],
+        [
+            'points' => [
+                ['coordinates' => ['x' => -1, 'y' => 1], 'rgb' => [0, 0, 0]],
+                ['coordinates' => ['x' => -2, 'y' => 2], 'rgb' => [255, 255, 255]],
+            ],
+        ],
+        [
+            'points' => [
+                ['coordinates' => ['x' => -13, 'y' => 13], 'rgb' => [-3, 258, 0]],
+                ['coordinates' => ['x' => -14, 'y' => 14], 'rgb' => [0, -4, 259]],
+            ],
+        ],
+    ],
+];
+$rule = Nested::rule([
+    'charts' => Each::rule(new Rules([
+        Nested::rule([
+            'points' => Each::rule(new Rules([
+                Nested::rule([
+                    'coordinates' => Nested::rule([
+                        'x' => [Number::rule()->min(-10)->max(10)],
+                        'y' => [Number::rule()->min(-10)->max(10)],
+                    ]),
+                    'rgb' => Each::rule(new Rules([
+                        Number::rule()->min(0)->max(255)->skipOnError(false),
+                    ])),
+                ])->skipOnError(false),
+            ])),
+        ])->skipOnError(false),
+    ])),
+]);
+$errors = $rule->validate($data)->getErrors();
+```
+
+The contents of the errors will be:
+
+```php
+$errors = [
+    'charts.0.points.0.coordinates.x' => 'Value must be no less than -10.',
+    'charts.0.points.0.rgb.0' => 'Value must be no less than 0. -1 given.',
```

Comment:
```suggestion
    'Value must be no less than -10.',
    'Value must be no less than 0. -1 given.',
```

Reply:
Thanks, but I think in this context it's better to change the called method.

Reply:
:heavy_check_mark: